### ISSUE TITLE
20-Arxiv-ReZero is All You Need: Fast Convergence at Large Depth

### ISSUE THREAD

**Issue by wangqiangneu** [wangqiangneu/MT-PaperReading#55]
## 简介
对残差连接的一种改进，能够更快的训练更深的网络。做法很简单。以transformer为例，移除layernorm，$y=x+a*F(x)$即可，其中`a`需要初始化成0。做法上很像`DLCL`，`DLCL`中也在每层中使用了learnable scalar，不同地方在于，`DLCL`强调的是跟以往更多层之间的关系，`ReZero`要学的参数是O(L)，`DLCL`学的是O(L^2)。另外，`ReZero`在transformer的每层中，所有`子层`都做这个scaled的操作，其中`a`是所有子层共享的。而`DLCL`是只在`层`中做，不包括子层。

## 论文信息
* Author: UCSD
* [Paper](https://arxiv.org/abs/2003.04887)
* [Code](https://github.com/majumderb/rezero)

## 总结
* 分析input-output的jacobian思路是对的，
* 方法很简单，可以作为baseline一试；去掉LN也很赞，多少能少算点嘛~
* 看repo里的issue，似乎有人直接用在[gpt](https://github.com/majumderb/rezero/issues/8)上是不好使的，会NAN；作者建议embedding初始化U(-1/d, +1/d)，所以用的时候可能还需要调整

**Comment by hukkai** [comment]
I think the idea of the paper can be found in Fixup initialization where the author explained why there would be some training divergence for scalar scale https://github.com/hongyi-zhang/Fixup/issues/6#issuecomment-506460413

**Comment by wangqiangneu** [comment]
> I think the idea of the paper can be found in Fixup initialization where the author explained why there would be some training divergence for scalar scale [hongyi-zhang/Fixup#6 (comment)](https://github.com/hongyi-zhang/Fixup/issues/6#issuecomment-506460413)

Good point! Maybe using a separated and small learning rate for those scalars is the key.
